soccer/gameplay/plays/offense/distraction.py

```diff
@@ -0,0 +1,178 @@
+        self.add_subbehavior(tactics.coordinated_pass.CoordinatedPass(self.center), 'center pass', required = True)
+        self.add_subbehavior(skills.move.Move(self.distraction_recieve_pass_point), 'move back to distract', required = False, priority = 10)
+
+    def on_exit_cetner_pass(self):
+        self.remove_all_subbehaviors()
+
+    def on_enter_passing(self):
+        #either pass to striker or distracter depending on shot chance        
+        pass_to_distract_chance = evaluation.passing.eval_pass( main.ball().pos, self.distraction_recieve_pass_point, main.our_robots() )
+        pass_to_striker_chance = evaluation.passing.eval_pass( main.ball().pos, self.striker_point, main.our_robots() )
+        shot_of_striker_chance1 = evaluation.shooting.eval_shot( self.striker_point, main.our_robots() )
+
+        if self.distract_box_max.x > main.ball().pos.x > self.distract_box_min.x and self.distract_box_max.y > main.ball().pos.y > self.distract_box_min.y:
+            #if the ball is already near the distracter, then no pass will occur and the distracter will just cpature the ball
+            self.add_subbehavior(skills.capture.Capture(), 'get close ball', required = True)
+        elif self.striker_box_max.x < main.ball().pos.x < self.striker_box_min.x and self.striker_box_max.y > main.ball().pos.y > self.striker_box_min.y:
+            #if the ball is already near the striker, then no pass will occur and the striker will just cpature the ball
+            self.add_subbehavior(skills.capture.Capture(), 'striker get close ball', required = True)
+        else:
+            if pass_to_distract_chance >= pass_to_striker_chance*shot_of_striker_chance1:
+                self.add_subbehavior(skills.move.Move(self.striker_point), 'make striker stay again', required = True)
+                self.add_subbehavior(tactics.coordinated_pass.CoordinatedPass(self.distraction_recieve_pass_point), 'distract pass', required = True)
+            else:
+                self.add_subbehavior(skills.move.Move(self.distraction_point), 'make striker stay again', required = True)
+                self.add_subbehavior(tactics.coordinated_pass.CoordinatedPass(self.striker_point), 'striker pass', required = True) 
+
+    def on_exit_passing(self):
+        self.remove_all_subbehaviors()
+
+    def on_enter_cross(self):
+        #if the ball is passed to the distractor the ball is passed to the striker, as the third robot moves to the right to distract more
+        pass_to_striker_chance = evaluation.passing.eval_pass(main.ball().pos, self.striker_point, main.our_robots() )
+        if self.striker_box_max.x < main.ball().pos.x < self.striker_box_min.x and self.striker_box_max.y > main.ball().pos.y > self.striker_box_min.y:
+            #if the ball is already near the striker, then no pass will occur and the striker will just cpature the ball
+            self.add_subbehavior(skills.capture.Capture(), 'capture 3', required = True)
```

Comment:
Does this happen ever too?

Reply:
Same as above, the setup state or something should be handling all the captures and just jump to the correct state depending on where it is

Reply:
This allows for the states to be more specific and closed in terms of scope without forcing other people to think too hard about all the situations in which this must work.